sunkit_image/tests/test_granule.py

```diff
@@ -0,0 +1,211 @@
+import random
+
+import numpy as np
+import pytest
+
+import sunpy
+import sunpy.map as sm
+
+import sunkit_image.granule as granule
+
+
+@pytest.fixture(scope="session")
+def inputs():
+    import warnings
+
+    with warnings.catch_warnings():
+        data_map = sunpy.map.Map("sunkit_image/tests/granule_testdata.fits")
+    test_res = 0.016
+    test_method = "li"
+    return data_map, test_res, test_method
+
+
+def test_segment(inputs):
+    """
+    Unit tests for segment() function.
+    """
+
+    data_map, test_res, test_method = inputs
+
+    # -------- positive tests -------- :
+    segmented = granule.segment(data_map, test_method, test_res, True)
+
+    # Test 1: check that the returned type is correct
+    assert type(segmented) is sunpy.map.mapbase.GenericMap
+    # Test 2: get an array of pixel values and check it is not empty
+    initial_pix = sm.all_pixel_indices_from_map(data_map).value
+    seg_pixels = sm.all_pixel_indices_from_map(segmented).value
+    assert np.size(seg_pixels) > 0
+    # Test 3: check that the returned shape is unchanged
+    assert seg_pixels.shape == initial_pix.shape
+
+    # -------- negative tests -------- :
+    # Test 4: check that the values in the array are changed
+    random.seed(42)
+    # pick 10 random indices to check
+    x = random.sample(list(np.arange(0, data_map.data.shape[0], 1)), 10)
+    y = random.sample(list(np.arange(0, data_map.data.shape[1], 1)), 10)
+    for i in range(len(x)):
+        assert data_map.data[x[i], y[i]] != segmented.data[x[i], y[i]]
+
+    # ------ error raising tests ------ :
+    # Test 5: check that errors are raised for incorrect inputs
+    with pytest.raises(TypeError, match="Input must be sunpy map."):
+        granule.segment(np.array([[1, 2, 3], [1, 2, 3]]), test_method, test_res)
+    with pytest.raises(TypeError, match="Method must be one of: "):
+        granule.segment(data_map, "banana", test_res)
+
+
+def test_get_threshold(inputs):
+    """
+    Unit tests for get_threshold() function.
+    """
+
+    _, _, test_method = inputs
+
+    # -------- positive tests -------- :
+    # Test 1: check type of output
+    test_arr1 = np.array([[1, 2, 3, 4, 5], [1, 2, 3, 4, 5]])
+    threshold1 = granule.get_threshold(test_arr1, test_method)
+    assert type(threshold1) is np.float64
+    # Test 2: check range of output
+    assert 0 < threshold1 < np.max(test_arr1)
+
+    # -------- negative tests -------- :
+    # Test 3: check that different arrays return different thresholds
+    test_arr2 = np.array([[2, 3, 4, 5, 6], [2, 3, 4, 5, 6]])
+    threshold2 = granule.get_threshold(test_arr2, test_method)
+    assert threshold1 != threshold2
+
+    # ------ error raising tests ------ :

```

Comment:
These should be a seperate unit test.

Reply:
Outside of test_get_threshold?

Reply:
Yes please.

Reply:
Sorry to for the continued confusion.. but if I were to place these lines in a seperate test, I would also need to place the block before it in that test, seeing as these lines compare threshold1 (created in the previous block) to threshold2 created here. Is that what you are envisioning?

Reply:
In this case, you would leave them combined. If you are testing the same output in different ways, that should be in the same unit test.